src/htmx.js

```diff
@@ -2848,10 +2848,10 @@ return (function () {
 
         function verifyPath(elt, path, requestConfig) {
             var url = new URL(path, document.location.href);
-            var hostname = document.location.hostname;
-            var sameHost = hostname !== url.hostname;
+            var origin = document.location.origin;
+            var sameHost = origin === url.origin;
```

Comment:
nitpick but I think this would be even simpler as:
```js
var sameHost = document.location.origin === url.origin
```
then it's a little clearer what you're comparing